ci/Jenkinsfile.groovy

```diff
@@ -569,25 +572,49 @@ ansiColor('xterm') {
-                            //
-                            //        At this point we'd also want to verify
-                            //        that the versions on all files are the
-                            //        same, and that if we are in release mode
-                            //        the version contains only digits and dots.
-                            //
+                            sh "ls dist/"
+
                             versionText = sh(script: """sed -ne "s/.*version='\\([^']*\\)',/\\1/p" src/datatable/_build_info.py""", returnStdout: true).trim()
                             sh "echo versionText = ${versionText}"
-                            def _versionText = versionText
+                            println "versionText = ${versionText}"
+
+                            def s3cmd = ""
+                            def pyindex_links = ""
+                            dir("dist") {
+                                def distDir = new File(".")
+                                distDir.traverse {
+                                    def sha256 = sh(script: """python -c "import hashlib;print(hashlib.sha256(open('${it.name}', 'rb').read()).hexdigest())" """,
+                                                    returnStdout: true).trim()
+                                    def s3path = "${S3_BASE}/dev/datatable-${versionText}/{it.name}"
+                                    s3cmd += "s3cmd put -P ${it.name} ${S3_BASE}/${s3path}\n"
+                                    pyindex_links += """  <li><a href="${S3_URL}/${s3path}#sha256=${sha256}">{it.name}</a></li>\n"""
+                                }
+                            }
+
+                            def pyindex_content = "${S3_URL}/index.html".toURL().text
+                            pyindex_content -= "</body></html>\n"
+                            pyindex_content += "<h2>${versionText}</h2>\n"
+                            pyindex_content += "<ul>\n"
+                            pyindex_content += pyindex_links
+                            pyindex_content += "</ul>\n"
+                            pyindex_content += "</body></html>\n"
+                            File pyindex_file = new File("index.html")
+                            pyindex_file.write(pyindex_content)
+                            s3cmd += "s3cmd put -P index.html ${S3_BASE}/index.html"
+
+                            docker.withRegistry("https://harbor.h2o.ai", "harbor.h2o.ai") {
+                                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "awsArtifactsUploader"]]) {
+                                    docker.image("harbor.h2o.ai/library/s3cmd").inside {
+                                        sh s3cmd
+                                    }
+                                }
+                            }
+
+                            // TODO: remove?
                             s3upDocker {
```

Comment:
regarding S3UpDocker:

The strategy which we are using in other projects:
1. prepare structure which we would like to see in the target bucket locally. E.g.:
```
dist/
dist/project.whl
dist/docs
dist/docs/index.html
dist/index.html
```
2. s3cmddocker is doing simple sync into target location defined by release/version/artifactid

Reply:
the lesson learned from that that we minimize code in Jenkinsfiles (which are hard to debug) and simply prepare the structure locally (which also enable some testing if necessary)

Reply:
I understand benefits of code sharing, but the problems with `s3upDocker` are:
- It is impossible to know (from reading the code) where the artifacts will end up. There is literally no S3 url in its parameters;
- It alters the paths of the files that it receives, splitting them into sub-directories. This is problematic because the index.html file needs to end up at the exact path, while other artifacts should be at some paths that are known beforehand;
- The function has neither documentation nor source code (at least GitHub cannot locate the source code: https://github.com/search?q=s3updocker&type=Code). It might be the permission issue.

At the same time, the alternative -- to construct the `s3cmd` command manually -- is perfectly simple, straightforward and understandable. The only code that would be useful is to reuse would be a function like
```
def s3docker(cmd) {
  docker.withRegistry("https://harbor.h2o.ai", "harbor.h2o.ai") {
    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "awsArtifactsUploader"]]) {
      docker.image("harbor.h2o.ai/library/s3cmd").inside {
        sh cmd
      }
    }
  }
}
```